other-identifiers/src/main/java/org/reactome/release/otherIdentifiers/PostStepChecks.java

```diff
@@ -0,0 +1,68 @@
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ *  This is a QA step that will be run after finishing an OtherIdentifiers run. For each species,
+ *  the count of ReferenceGeneProduct instances that have their 'otherIdentifier' attribute filled is found
+ *  between the current and previous database. If the previous database has a higher count, a warning message
+ *  is output and should be looked into. If the count is still relatively close,  it can probably be ignored.
+ */
+
+public class PostStepChecks {
+    private static final Logger logger = LogManager.getLogger();
+
+    public static void compareOtherIdentifierCounts(MySQLAdaptor dba, MySQLAdaptor dbaPrev) throws Exception {
+
+        Collection<GKInstance> speciesInstances = dba.fetchInstancesByClass(ReactomeJavaConstants.Species);
+
+        for (GKInstance speciesInstance : speciesInstances) {
+            int currentOtherIdentifierCount = getRGPInstanceOtherIdentifierCount(dba, speciesInstance);
+            int previousOtherIdentifierCount = getRGPInstanceOtherIdentifierCount(dbaPrev, speciesInstance);
+            if (currentOtherIdentifierCount < previousOtherIdentifierCount) {
+                logger.warn(speciesInstance.getDisplayName() + " has fewer ReferenceGeneProduct instances with OtherIdentifiers compared to previous release:  " +
+                        "release_current - " + currentOtherIdentifierCount + " -- release_previous - " + previousOtherIdentifierCount);
+            }
+        }
+    }
+
+    /**
+     * Wrapper method for determining other identifier count.
+     * @param dba MySQLAdaptor
+     * @param speciesInstance GKInstance pertaining to a specific species
+     * @return The other identifier count, as an int.
+     * @throws Exception
+     */
+    private static int getRGPInstanceOtherIdentifierCount(MySQLAdaptor dba, GKInstance speciesInstance) throws Exception {
+        Collection<GKInstance> rgpInstances = dba.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceGeneProduct, ReactomeJavaConstants.species, "=", speciesInstance);
+
+        int currentOtherIdentifierCount = getOtherIdentifierCount(rgpInstances);
+        return currentOtherIdentifierCount;
+    }
+
+    /**
+     * Determines the number of ReferenceGeneProduct instances that have a filled otherIdentifier attribute.
+     * @param rgpInstances Collection of RGP GKInstances associated with a species.
+     * @return An int is returned, representing the otherIdentifier count.
+     */
+    public static int getOtherIdentifierCount(Collection<GKInstance> rgpInstances) {
+        AtomicInteger currentOtherIdentifierCount = new AtomicInteger();
+        rgpInstances.forEach(rgpInstance -> {
+            try {
+                if (rgpInstance.getAttributeValuesList(ReactomeJavaConstants.otherIdentifier).size() > 0) {
```

Comment:
You may need to check that the value you get back is not null before calling `.size()` to ensure there is no NullPointerException being thrown.

Reply:
Since its `getAttributeValuesList` it will return an empty array here. I believe it would only return null if 'otherIdentifier' wasn't a valid attribute for the instance type

Reply:
I hope that would throw an exception to break the application, since that really should not happen.

Reply:
Your correct, I checked afterwards and if it isn't a valid attribute then an exception is thrown

Reply:
And it causes the application to terminate? The InvalidAttributeException (I think) doesn't get swallowed/suppressed?

Reply:
It doesn't cause it to terminate since the error is caught in the try/catch. In this case when I tried it it just caused continuous stacktraces to be output

Reply:
Oh. I'm thinking that this type of exception _should_ break the application flow.

Reply:
The approach in this comment might help - https://github.com/reactome/data-release-pipeline/pull/113#discussion_r344835986.  If an exception happens, it probably does indicate something unusual we should investigate.

Reply:
From the perspective of my code, I am explicitly pulling out `ReferenceGeneProduct` instances. Based on this, there isn't any concern about this exception being thrown unless the Data Model changes

Reply:
But I think the code requires the exception to be handled in order to compile?

Reply:
It does. It gets caught, and the stack trace gets printed.

Reply:
True, but if the exception is being caught in the code anyway and if it were to occur, termination of the program is probably the best way to go.